File: packages/csharp/src/Anip.Service/SignedManifest.cs
namespace Anip.Service;

/// <summary>
/// Holds a manifest's canonical JSON bytes and its detached JWS signature.
/// </summary>
public class SignedManifest
{
    /// <summary>The canonical JSON bytes of the manifest.</summary>
    public byte[] ManifestJson { get; }

    /// <summary>The detached JWS signature (header..signature).</summary>
    public string Signature { get; }

    public SignedManifest(byte[] manifestJson, string signature)
    {
        ManifestJson = manifestJson;
        Signature = signature;
    }
}

File: packages/java/anip-service/src/main/java/dev/anip/service/SignedManifest.java
package dev.anip.service;

/**
 * Holds the manifest as canonical JSON bytes and its detached JWS signature.
 */
public class SignedManifest {

    private final byte[] manifestJson;
    private final String signature;

    public SignedManifest(byte[] manifestJson, String signature) {
        this.manifestJson = manifestJson;
        this.signature = signature;
    }

    public byte[] getManifestJson() {
        return manifestJson;
    }

    public String getSignature() {
        return signature;
    }
}
